pkg/api/error.go

```diff
@@ -116,6 +116,12 @@ func WriteError(w http.ResponseWriter, statusCode int, code, target, message str
 
 // WriteCloudError writes a CloudError to the given ResponseWriter
 func WriteCloudError(w http.ResponseWriter, err *CloudError) {
+	if err.StatusCode == http.StatusInternalServerError {
+		crID := w.Header().Get("X-Ms-Correlation-Request-Id")
+		if crID != "" {
+			err.Message += " (correlationId: " + crID + ")"
+		}
+	}
```

Comment:
I've lost the plot on this PR.  @mjudeikis can you help?

I have just validated using the current prod RP that ARM already injects x-ms-correlation-request-id in the return path for us, and due to [this az functionality](https://github.com/Azure/azure-cli/blob/c3982c0070ae2df31ec319265f991c88c17748e8/src/azure-cli-core/azure/cli/core/commands/arm.py#L120-L121), you will see errors like "Deployment failed. Correlation ID: e8ba3d1f-da3c-4c39-9a88-5ad3234209dd. The provided service principal credentials are invalid."

What else is being requested?
Why would we want this hunk?

Reply:
So these are for Deployment Failed. We need tracking with "InternalServerError` errors. If this is already present - we good. But need to check.